pandas/core/dtypes/cast.py

```diff
@@ -110,7 +110,9 @@ def trans(x):  # noqa
                     np.prod(result.shape)):
                 return result
 
-        if issubclass(dtype.type, np.floating):
+        # don't convert bool to float GH16875
+        if (issubclass(dtype.type, np.floating) and
+                not is_bool_dtype(result.dtype)):
             return result.astype(dtype)
         elif is_bool_dtype(dtype) or is_integer_dtype(dtype):
```

Comment:
try making this the 2nd if (iow move if bool/int) first

not sure will work but a bit cleaner